samples/sensor/lis2dw12/src/main.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2024 Pierrick Curt
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+
+static void fetch_and_display(const struct device *sensor)
+{
+	static unsigned int count;
+	struct sensor_value accel[3];
+	struct sensor_value temperature;
+	int rc;
+
+	rc = sensor_sample_fetch_chan(sensor, SENSOR_CHAN_ACCEL_XYZ);
+	if (rc < 0) {
+		printf("ERROR: SENSOR_CHAN_ACCEL_XYZ fetch failed: %d\n", rc);
+	}
+	rc = sensor_sample_fetch_chan(sensor, SENSOR_CHAN_DIE_TEMP);
+	if (rc < 0) {
+		printf("ERROR: SENSOR_CHAN_DIE_TEMP fetch failed: %d\n", rc);
+	}
+
+	++count;
+	if (rc == 0) {
+		rc = sensor_channel_get(sensor, SENSOR_CHAN_ACCEL_XYZ, accel);
+		if (rc == 0) {
+			rc = sensor_channel_get(sensor, SENSOR_CHAN_DIE_TEMP, &temperature);
+		}
+	}
+	if (rc < 0) {
+		printf("ERROR: Update failed: %d\n", rc);
+	} else {
+		printf("#%u @ %u ms: x %f , y %f , z %f, t %f\n", count, k_uptime_get_32(),
+		       sensor_value_to_double(&accel[0]), sensor_value_to_double(&accel[1]),
+		       sensor_value_to_double(&accel[2]), sensor_value_to_double(&temperature));
```

Comment:
sensor_balue_to_double() returns a double. Is it ok printing a float (%f)?

Reply:
Yes, printf function automatically converts float in double, so it's ok here.

Reply:
ok